packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -326,6 +328,12 @@ class StdoutLogger extends Logger {
 
   @override
   void sendEvent(String name, [Map<String, dynamic> args]) { }
+
+  @override
+  void clear() {
+    _clearStatus();
```

Comment:
Clearing the status looks like it will confuse the check for nested status spinners above on line 290. Since we can't cancel a status already returned from `startProgress`, we probably have to keep the status when clearing the screen.

If this creates a mess on the screen for some reason, you can look at the way `printStatus()` works:
```dart
_status?.pause();
// writes stuff
_status?.resume();
```

Reply:
Do you mean if the user tries to clear while a spinner is on screen? I suppose that since keyboard inputs are ignored during loading anyway (like hot restart and reload), `_clearStatus` is probably unnecessary.

Reply:
A future user of the API might try to call `Logger.clear()` while a spinner is active. I don't know whether that would cause the spinner too misbehave, but in the other methods where we write to stdout/stderr, we first pause the status spinner, so we should probably follow suit here.

Reply:
Good point, I fixed it.